src/app/georeferencer/qgsgeorefmainwindow.h

```diff
@@ -237,7 +253,9 @@ class QgsGeoreferencerMainWindow : public QMainWindow, private Ui::QgsGeorefPlug
     QgsGCPList mPoints;
     QgsGCPList mInitialPoints;
     QgsMapCanvas *mCanvas = nullptr;
-    std::unique_ptr< QgsRasterLayer > mLayer;
+    std::unique_ptr< QgsRasterLayer > mRLayer;
+    std::unique_ptr< QgsVectorLayer > mVLayer;
```

Comment:
```suggestion
    std::unique_ptr< QgsMapLayer > mLayer;
```

The code could be simplified a lot by using a single member for layer storage

Reply:
I thought that always recasting would be worst than storing a single pointer per type.

Reply:
done